packages/elements/src/react/common/loading.tsx

```diff
@@ -157,15 +156,15 @@ function SignInLoading({ children, scope, routerRef }: SignInLoadingProps) {
     computedScope = inferredScope;
   }
 
-  const isChooseStrategyStep = useActiveTags(
-    routerRef,
-    ['route:first-factor', 'route:choose-strategy'],
-    ActiveTagsMode.all,
-  );
+  const snapshot = routerRef.getSnapshot();
+  const isFirstFactor = snapshot.hasTag('route:first-factor');
+
   // Determine loading states based on the step
   const isStartLoading = isLoading && loadingStep === 'start';
   const isVerificationsLoading = isLoading && loadingStep === 'verifications';
-  const isChooseStrategyLoading = isLoading && isChooseStrategyStep;
+  const isChooseStrategyLoading = isLoading && isFirstFactor && snapshot.hasTag('route:choose-strategy');
+  const isForgotPasswordLoading = isFirstFactor && snapshot.hasTag('route:forgot-password');
+  const isResetPasswordLoading = isFirstFactor && snapshot.hasTag('route:reset-password');
```

Comment:
Nit: `isLoading` check is missing (since we do that in the other boolean constants, too)